dash.el

```diff
@@ -2274,9 +2274,22 @@ The test for equality is done with `equal',
 or with `-compare-fn' if that's non-nil.
 
 Alias: `-uniq'"
-  (let (result)
-    (--each list (unless (-contains? result it) (!cons it result)))
-    (nreverse result)))
+  ;; Implementation note: Using a hash-table to remove duplicates is
+  ;; considerably faster than do it by iteration when the length
+  ;; of the list is greater than 32.
```

Comment:
Nitpick: I think "iteration" is a misnomer here. Both the hash-based and list-based approaches are iterative. The difference is in the data structure used as a lookup table. So I suggest the following wording instead:

```el
  ;; Implementation note: The speedup gained from hash table lookup
  ;; starts to outweigh its overhead for lists of length greater than
  ;; 32.  See discussion in PR #305.
```